xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -9499,6 +9529,27 @@ public DocumentAuthors getAuthors()
         return this.authors;
     }
 
+    @Override
+    public RequiredRights getRequiredRights()
+    {
+        return this.requiredRights;
+    }
+
+    /**
+     * Set {@link #isMetaDataDirty} if the required rights are changed.
+     *
+     * @param requiredRights the required rights to set
+     * @since 15.3RC1
+     */
+    @Unstable
+    public void setRequiredRights(RequiredRights requiredRights)
```

Comment:
It's not very clear to me how someone can modify the required rights since there is no public setters for them (`DefaultRequiredRights` is internal).

Why don't you simply follow the same logic as DocumentAuthors since it's a very similar use case?

Reply:
Because `DocumentAuthors` is only relying on fields of XWikiDocument itself. Whereas, I need to access values of an external table.

Reply:
> Because DocumentAuthors is only relying on fields of XWikiDocument itself. Whereas, I need to access values of an external table.

I don't see the relationship with making RequiredRights readonly.

In any case, the current API makes it impossible to set required rights from what I understand.

Reply:
OK, this whole conversation does not make any sense because I missed RequiredRights#setRights.